### PR TITLE
Nutrition phase test

### PR DESCRIPTION
**Pull request by ninsid711**
Made changes to the prompt, parameters and test files

### CODE REVIEW

File: services/nutrition/controllers/diet_gen.go [hunk 1]
```diff
@@ -57,9 +61,8 @@ func validateSubstituteParams(params SubstituteParams) error {
...
 	// Check for invalid or missing values
 	for _, v := range []string{
-		params.Plan, params.Activity, params.TargetCal, params.TargetProtein,
-		params.TargetFat, params.TargetCarbs, params.Cuisine, params.MealChoice,
-		params.Occupation, params.Allergies, params.OtherPreferences, params.Variety, params.Budget,
+		params.Gender, params.Goal, params.ActivityLevel, params.Cuisine, params.MealChoice,
+		params.Allergies, params.OtherPreferences, params.Variety,
 	} {
 		if v == "" || v == "unknown" {
 			return errors.New("some parameters are invalid or missing")
```
Comment (achyuthcodes30) [inline services/nutrition/controllers/diet_gen.go:68]: Not happy with how validation is being done here, please make it stricter and cover all edge cases. Some optional preferences like allergies and other preferences can be empty and mealChoice can be empty too and others like BMI, duration, meals per day cannot be empty or must atleast have a default!

**Review by achyuthcodes30 — CHANGES_REQUESTED** [review]

File: services/nutrition/controllers/diet_gen.go (hunk 1 above)
Comment (ninsid711) [reply services/nutrition/controllers/diet_gen.go:68]: Nihal had written those tests, but I'll make the changes for covering all the edge cases.

**Review by achyuthcodes30 — APPROVED** [review]
